src/Form/Input.js

```diff
@@ -241,7 +241,7 @@ class Input extends Component {
       ...rest
     } = this.props;
 
-    const isFloating = (floating && value && value.length > 0) || (floating && this.state.value);
+    const isFloating = floating && ((value && value.length > 0) || (this.state.value || this.state.value === 0));
```

Comment:
A bit confused on why there is a comparison against both the `props.value` and `state.value`.  Shouldn't we take in the `props.value` and set `state.value` during initialization (or props updates) and only refer to the `state.value` throughout the component, since that will always hold the current input value?

Reply:
I agree here. `state.value` is kept in sync with `props.value` via `getDerivedStateFromProps`. I think the condition can be summed as ```floating && value != undefined && `${value}`.length > 0```. Note we're not doing strict equality here so we capture both null and undefined values since `null == undefined`. The template string will coalesce a number into a string so we can check the length property.

Looking at you https://github.com/tc39/proposal-nullish-coalescing to make this easier